packages/purpose-process/src/services/purposeService.ts

```diff
@@ -409,6 +412,92 @@ export function purposeServiceBuilder(
       await repository.createEvent(event);
       return archivedVersion;
     },
+
+    async suspendPurposeVersion({
+      purposeId,
+      versionId,
+      organizationId,
+      correlationId,
+    }: {
+      purposeId: PurposeId;
+      versionId: PurposeVersionId;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<PurposeVersion> {
+      logger.info(`Suspending Version ${versionId} in Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+
+      const eservice = await retrieveEService(
+        purpose.data.eserviceId,
+        readModelService
+      );
+
+      const suspender = getOrganizationRole({
+        organizationId,
+        producerId: eservice.producerId,
+        consumerId: purpose.data.consumerId,
+      });
+
+      const purposeVersion = retrievePurposeVersion(versionId, purpose);
+
+      if (
+        purposeVersion.state !== purposeVersionState.active &&
+        purposeVersion.state !== purposeVersionState.suspended
+      ) {
+        throw notValidVersionState(purposeVersion.id, purposeVersion.state);
+      }
+
+      const suspendedPurposeVersion: PurposeVersion = {
+        ...purposeVersion,
+        state: purposeVersionState.suspended,
+        suspendedAt: new Date(),
+        updatedAt: new Date(),
+      };
+
+      const event = match(suspender)
+        .with(ownership.CONSUMER, () => {
+          const updatedPurpose: Purpose = {
+            ...replacePurposeVersion(purpose.data, suspendedPurposeVersion),
+            suspendedByConsumer: true,
+          };
+          return toCreateEventPurposeSuspendedByConsumer({
+            purpose: updatedPurpose,
+            purposeVersionId: versionId,
+            version: purpose.metadata.version,
+            correlationId,
+          });
+        })
+        .with(ownership.PRODUCER, () => {
+          const updatedPurpose: Purpose = {
+            ...replacePurposeVersion(purpose.data, suspendedPurposeVersion),
+            suspendedByProducer: true,
+          };
+          return toCreateEventPurposeSuspendedByProducer({
+            purpose: updatedPurpose,
+            purposeVersionId: versionId,
+            version: purpose.metadata.version,
+            correlationId,
+          });
+        })
+        .with(ownership.SELF_CONSUMER, () => {
+          const updatedPurpose: Purpose = {
+            ...replacePurposeVersion(purpose.data, suspendedPurposeVersion),
+            suspendedByConsumer: true,
+            suspendedByProducer: true,
+          };
+          return toCreateEventPurposeSuspendedByConsumer({
+            purpose: updatedPurpose,
+            purposeVersionId: versionId,
+            version: purpose.metadata.version,
+            correlationId,
+          });
+        })
```

Comment:
```suggestion
```
In the current implementation there's no distinction between `SELF_CONSUMER` and `PRODUCER` in this case, unless the business logic isn't changed we should treat that case like the `PRODUCER`

Reply:
Updated